In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#Load the datasets

In [3]:
orders = pd.read_csv(r"C:\Users\USER\Downloads/orders.csv")

order_details = pd.read_csv(r"C:\Users\USER\Downloads/order_details.csv")

pizzas = pd.read_csv(r"C:\Users\USER\Downloads/pizzas.csv")

pizza_types = pd.read_csv(r"C:\Users\USER\Downloads/pizza_types.csv", encoding='Unicode_escape')

In [4]:
#Merge into one dataframe
df = pd.merge(orders, order_details, on='order_id')

df = pd.merge(df, pizzas, on='pizza_id')

df = pd.merge(df, pizza_types, on='pizza_type_id')

In [5]:
df.head()

order_id        date      time  order_details_id       pizza_id  quantity  \
0         1  2015-01-01  11:38:36                 1     hawaiian_m         1   
1         2  2015-01-01  11:57:40                 2  classic_dlx_m         1   
2         2  2015-01-01  11:57:40                 3  five_cheese_l         1   
3         2  2015-01-01  11:57:40                 4    ital_supr_l         1   
4         2  2015-01-01  11:57:40                 5     mexicana_m         1   

  pizza_type_id size  price                       name category  \
0      hawaiian    M  13.25         The Hawaiian Pizza  Classic   
1   classic_dlx    M  16.00   The Classic Deluxe Pizza  Classic   
2   five_cheese    L  18.50      The Five Cheese Pizza   Veggie   
3     ital_supr    L  20.75  The Italian Supreme Pizza  Supreme   
4      mexicana    M  16.00         The Mexicana Pizza   Veggie   

                                         ingredients  
0           Sliced Ham, Pineapple, Mozzarella Cheese  
1  Pepperoni, Mushrooms, Red Onions, Red Peppers,...  
2  Mozzarella Cheese, Provolone Cheese, Smoked Go...  
3  Calabrese Salami, Capocollo, Tomatoes, Red Oni...  
4  Tomatoes, Red Peppers, Jalapeno Peppers, Red O...

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48620 entries, 0 to 48619
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   order_id          48620 non-null  int64  
 1   date              48620 non-null  object 
 2   time              48620 non-null  object 
 3   order_details_id  48620 non-null  int64  
 4   pizza_id          48620 non-null  object 
 5   quantity          48620 non-null  int64  
 6   pizza_type_id     48620 non-null  object 
 7   size              48620 non-null  object 
 8   price             48620 non-null  float64
 9   name              48620 non-null  object 
 10  category          48620 non-null  object 
 11  ingredients       48620 non-null  object 
dtypes: float64(1), int64(3), object(8)
memory usage: 4.5+ MB


In [7]:
#Ensure date and time columns are in datetime format

df['date'] = pd.to_datetime(df['date'])

df['time'] = pd.to_datetime(df['time'])

C:\Users\USER\AppData\Local\Temp\ipykernel_11856\2150434812.py:5: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['time'] = pd.to_datetime(df['time'])


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48620 entries, 0 to 48619
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   order_id          48620 non-null  int64         
 1   date              48620 non-null  datetime64[ns]
 2   time              48620 non-null  datetime64[ns]
 3   order_details_id  48620 non-null  int64         
 4   pizza_id          48620 non-null  object        
 5   quantity          48620 non-null  int64         
 6   pizza_type_id     48620 non-null  object        
 7   size              48620 non-null  object        
 8   price             48620 non-null  float64       
 9   name              48620 non-null  object        
 10  category          48620 non-null  object        
 11  ingredients       48620 non-null  object        
dtypes: datetime64[ns](2), float64(1), int64(3), object(6)
memory usage: 4.5+ MB


In [9]:
#To check for total revenue or sales

df["revenue"] = df["price"] * df["quantity"]

In [10]:
#Q1: Total Revenue/Sales

total_revenue = df["revenue"].sum()
print(f"1. Total Revenue: ${total_revenue: ,.2f}")

1. Total Revenue: $ 817,860.05


In [11]:
#Q2: Find the total quantity sold

total_quantity = df["quantity"].sum()
print(f"2 Total Quantity Sold: {total_quantity:,}")

2 Total Quantity Sold: 49,574


The total revenue for the year was $ 817,860.05, driven by 49,574 units sold. This provides a baseline for evaluating the store's annual performance and scaling potential.

In [12]:
#Q3 Find the total orders

total_orders = df["order_id"].nunique()
print(f"3 Total Orders: {total_orders:,}")

3 Total Orders: 21,350


In [13]:
#Q4 How many pizza types do they sell

pizza_type = df["name"].nunique()
print(f"4 Total Pizza Types: {pizza_type}")

4 Total Pizza Types: 32


In [14]:
#Q5 Find the average price of the pizzas

avg_pizza_price = df["price"].mean()
print(f"5 Average Price of Pizzas: {avg_pizza_price: .2f}")

5 Average Price of Pizzas:  16.49


With 49,574 unique orders, we can calculate that the average customer spend is approximately 16.50. The average menu price of $16.49 suggests a mid-range pricing strategy


In [15]:
#Q6 What are the peak hours of sales

df['hour'] = df['time'].dt.hour

peak_hours = df.groupby('hour')['order_id'].nunique().sort_values(ascending=False)

print("\n 6. The Peak Hour of Sales (Top 3): ")
print(peak_hours.head(3))


 6. The Peak Hour of Sales (Top 3): 
hour
12    2520
13    2455
18    2399
Name: order_id, dtype: int64


In [16]:
#Q7 Total sales made on each day of the week

df['day_of_week'] = df['date'].dt.day_name()

sales_by_day = df.groupby('day_of_week')['revenue'].sum().sort_values(ascending=False)

print("7. Sales by Day of the Week:")


print(sales_by_day)

7. Sales by Day of the Week:
day_of_week
Friday       136073.90
Thursday     123528.50
Saturday     123182.40
Wednesday    114408.40
Tuesday      114133.80
Monday       107329.55
Sunday        99203.50
Name: revenue, dtype: float64


In [17]:
#Q8 Top 5 best selling pizzas (by quantity) 

top_5_pizzas = df.groupby('name')['quantity'].sum().sort_values(ascending=False)

print("5. The top 5 best selling pizzas are:")

print(top_5_pizzas.head())

5. The top 5 best selling pizzas are:
name
The Classic Deluxe Pizza      2453
The Barbecue Chicken Pizza    2432
The Hawaiian Pizza            2422
The Pepperoni Pizza           2418
The Thai Chicken Pizza        2371
Name: quantity, dtype: int64


In [18]:
#Q9 Sales made in each month

df['month']= df['date'].dt.month_name()

sales_in_each_month = df.groupby('month')['revenue'].sum().sort_values(ascending=False)

sales_in_each_month

month
July         72557.90
May          71402.75
March        70397.10
November     70395.35
January      69793.30
April        68736.80
August       68278.25
June         68230.20
February     65159.60
December     64701.15
September    64180.05
October      64027.60
Name: revenue, dtype: float64

### Insight: Monthly Sales Trends
The analysis reveals that the pizza place maintains a steady stream of income throughout the year, with a narrow variance between the highest and lowest-earning months.

1. **Top Performance:** July is the most profitable month, followed closely by May. This may correlate with social gatherings or events during the summer season.

2. **Lowest Performance:** Sales dip slightly in September and October.

3. **Observation:** The differences between the highest month (July) and the lowest month (October) is only about $8,500, showing that the business is not heavily seasonal and enjoys stable demand year-round.

In [19]:
#Q10 Pizzas not doing well (bottom 5 by quantity)

bottom_5_pizzas = df.groupby('name')['quantity'].sum().sort_values(ascending=True).head(5)

print("\nBottom 5 Pizzas (Potential Menu Removal):")
print(bottom_5_pizzas)


Bottom 5 Pizzas (Potential Menu Removal):
name
The Brie Carre Pizza         490
The Mediterranean Pizza      934
The Calabrese Pizza          937
The Spinach Supreme Pizza    950
The Soppressata Pizza        961
Name: quantity, dtype: int64


In [22]:
df

order_id       date                time  order_details_id  \
0             1 2015-01-01 2026-02-18 11:38:36                 1   
1             2 2015-01-01 2026-02-18 11:57:40                 2   
2             2 2015-01-01 2026-02-18 11:57:40                 3   
3             2 2015-01-01 2026-02-18 11:57:40                 4   
4             2 2015-01-01 2026-02-18 11:57:40                 5   
...         ...        ...                 ...               ...   
48615     21348 2015-12-31 2026-02-18 21:23:10             48616   
48616     21348 2015-12-31 2026-02-18 21:23:10             48617   
48617     21348 2015-12-31 2026-02-18 21:23:10             48618   
48618     21349 2015-12-31 2026-02-18 22:09:54             48619   
48619     21350 2015-12-31 2026-02-18 23:02:05             48620   

            pizza_id  quantity pizza_type_id size  price  \
0         hawaiian_m         1      hawaiian    M  13.25   
1      classic_dlx_m         1   classic_dlx    M  16.00   
2      five_cheese_l         1   five_cheese    L  18.50   
3        ital_supr_l         1     ital_supr    L  20.75   
4         mexicana_m         1      mexicana    M  16.00   
...              ...       ...           ...  ...    ...   
48615  ckn_alfredo_m         1   ckn_alfredo    M  16.75   
48616  four_cheese_l         1   four_cheese    L  17.95   
48617   napolitana_s         1    napolitana    S  12.00   
48618     mexicana_l         1      mexicana    L  20.25   
48619      bbq_ckn_s         1       bbq_ckn    S  12.75   

                             name category  \
0              The Hawaiian Pizza  Classic   
1        The Classic Deluxe Pizza  Classic   
2           The Five Cheese Pizza   Veggie   
3       The Italian Supreme Pizza  Supreme   
4              The Mexicana Pizza   Veggie   
...                           ...      ...   
48615   The Chicken Alfredo Pizza  Chicken   
48616       The Four Cheese Pizza   Veggie   
48617        The Napolitana Pizza  Classic   
48618          The Mexicana Pizza   Veggie   
48619  The Barbecue Chicken Pizza  Chicken   

                                             ingredients  revenue  hour  \
0               Sliced Ham, Pineapple, Mozzarella Cheese    13.25    11   
1      Pepperoni, Mushrooms, Red Onions, Red Peppers,...    16.00    11   
2      Mozzarella Cheese, Provolone Cheese, Smoked Go...    18.50    11   
3      Calabrese Salami, Capocollo, Tomatoes, Red Oni...    20.75    11   
4      Tomatoes, Red Peppers, Jalapeno Peppers, Red O...    16.00    11   
...                                                  ...      ...   ...   
48615  Chicken, Red Onions, Red Peppers, Mushrooms, A...    16.75    21   
48616  Ricotta Cheese, Gorgonzola Piccante Cheese, Mo...    17.95    21   
48617  Tomatoes, Anchovies, Green Olives, Red Onions,...    12.00    21   
48618  Tomatoes, Red Peppers, Jalapeno Peppers, Red O...    20.25    22   
48619  Barbecued Chicken, Red Peppers, Green Peppers,...    12.75    23   

      day_of_week     month  
0        Thursday   January  
1        Thursday   January  
2        Thursday   January  
3        Thursday   January  
4        Thursday   January  
...           ...       ...  
48615    Thursday  December  
48616    Thursday  December  
48617    Thursday  December  
48618    Thursday  December  
48619    Thursday  December  

[48620 rows x 16 columns]